src/test/regress/expected/single_shard_table_udfs.out

```diff
@@ -0,0 +1,207 @@
+---------------------------------------------------------------------
+
+(1 row)
+
+-- should return a single element array that only includes its own shard id
+SELECT shardid=unnest(get_colocated_shard_array(shardid))
+FROM (SELECT shardid FROM pg_dist_shard WHERE logicalrelid='null_dist_key_table'::regclass) as shardid;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+BEGIN;
+  SELECT master_remove_partition_metadata('null_dist_key_table'::regclass::oid, 'null_dist_key_udfs', 'null_dist_key_table');
+ master_remove_partition_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+  -- should print 0
+  select count(*) from pg_dist_partition where logicalrelid='null_dist_key_table'::regclass;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+ROLLBACK;
+-- should fail --
+SELECT update_distributed_table_colocation('null_dist_key_table', colocate_with => 'none');
+ERROR:  relation null_dist_key_table should be a hash distributed table
+SELECT master_create_empty_shard('null_dist_key_table');
+ERROR:  relation "null_dist_key_table" is a single shard table
+DETAIL:  We currently don't support creating shards on single shard tables
+-- return true
+SELECT citus_table_is_visible('null_dist_key_table'::regclass::oid);
+ citus_table_is_visible
+---------------------------------------------------------------------
+ t
+(1 row)
+
+-- return false
+SELECT relation_is_a_known_shard('null_dist_key_table');
+ relation_is_a_known_shard
+---------------------------------------------------------------------
+ f
+(1 row)
+
+-- return | false | true |
+SELECT citus_table_is_visible(tableName::regclass::oid), relation_is_a_known_shard(tableName::regclass)
+FROM (SELECT tableName FROM pg_catalog.pg_tables WHERE tablename LIKE 'null_dist_key_table%') as tableName;
+ citus_table_is_visible | relation_is_a_known_shard
+---------------------------------------------------------------------
+ t                      | f
+(1 row)
+
+-- should fail, maybe support in the future
+SELECT create_reference_table('null_dist_key_table');
+ERROR:  table "null_dist_key_table" is already distributed
+SELECT create_distributed_table('null_dist_key_table', 'a');
+ERROR:  table "null_dist_key_table" is already distributed
+SELECT create_distributed_table_concurrently('null_dist_key_table', 'a');
+ERROR:  table "null_dist_key_table" is already distributed
+SELECT citus_add_local_table_to_metadata('null_dist_key_table');
+ERROR:  table "null_dist_key_table" is already distributed
+-- test altering distribution column, fails for single shard tables
+SELECT alter_distributed_table('null_dist_key_table', distribution_column := 'a');
+ERROR:  relation null_dist_key_table should be a hash distributed table
+-- test altering shard count, fails for single shard tables
+SELECT alter_distributed_table('null_dist_key_table', shard_count := 6);
```

Comment:
Another question: Should we allow setting shard key of a distributed table to NULL, in order to convert it into a single shard table?

Even though this wouldn't be useful for schema based sharding, it would be weird to disallow that in a world where create_distributed_table allows specifying shard key to be NULL.